packages/core/react/use-chat.ts

```diff
@@ -205,30 +206,37 @@ export function useChat({
   generateId = nanoid,
 }: Omit<UseChatOptions, 'api'> & {
   api?: string | StreamingReactResponseAction;
+  key?: string;
 } = {}): UseChatHelpers {
   // Generate a unique id for the chat if not provided.
   const hookId = useId();
-  const chatId = id || hookId;
+  const chatKey = [key, id ?? hookId];
 
   // Store a empty array as the initial messages
   // (instead of using a default parameter value that gets re-created each time)
   // to avoid re-renders:
   const [initialMessagesFallback] = useState([]);
 
   // Store the chat state in SWR, using the chatId as the key to share states.
-  const { data: messages, mutate } = useSWR<Message[]>([api, chatId], null, {
-    fallbackData: initialMessages ?? initialMessagesFallback,
-  });
+  const { data: messages, mutate } = useSWR<Message[]>(
+    [...chatKey, 'messages'],
```

Comment:
No need to spread, SWR uses deep comparison to track key changes:

```suggestion
    [chatKey, 'messages'],
```